### PR TITLE
The new .NET project file format support

### PR DESCRIPTION
**Pull request by nikitozz**
### Description of the PR

Adding support for the new .NET project files format (introduced in VS 2017, https://docs.microsoft.com/en-us/dotnet/articles/core/tools/project-json-to-csproj) for generated C# clients.

The feature is disabled by default and can be enabled by using `netCoreProjectFile` flag when generating a C# client. For example: `java -jar ./modules/swagger-codegen-cli/target/swagger-codegen-cli.jar generate -i modules/swagger-codegen/src/test/resources/2_0/petstore-with-fake-endpoints-models-for-testing.yaml -l csharp -o samples/client/petstore/csharp/SwaggerClientNetCoreProject --additional-properties netCoreProjectFile=true`.

New sample scripts are added and a new sample is generated.

### CODE REVIEW

**Comment by wing328** [comment]
@nikitozz thanks for the enhancement. We'll review and let you know if we've any question.

cc @jimschubert

File: bin/csharp-petstore-netcore-project.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/bin/sh
```
Comment (jimschubert) [inline bin/csharp-petstore-netcore-project.sh]: This file isn't executable.

Need to run:

```
chmod u+x bin/csharp-petstore-netcore-project.sh
```

**Comment by jimschubert** [comment]
I don't have VS 2017, so I can't verify this in the IDE right now. I did build and generate locally, then ran `dotnet restore` followed by `dotnet build` and things seem to be as expected.

File: bin/csharp-petstore-netcore-project.sh (hunk 1 above)
Comment (wing328) [inline bin/csharp-petstore-netcore-project.sh]: I'll take care of that.

**Comment by wing328** [comment]
@jimschubert thanks for reviewing the change.

@nikitozz thanks for the contribution.

**Comment by wing328** [comment]
FYI. Made 4 shell scripts executable via 1cd8341

**Pull request merged**